c_api/cmake/FindTaichi.cmake

```diff
@@ -0,0 +1,172 @@
+#[=======================================================================[.rst:
+FindTaichi
+----------
+
+Finds the Taichi library.
+
+The module first attempts to locate ``TaichiConfig.cmake`` in any Taichi
+installation in CMake variable ``TAICHI_C_API_INSTALL_DIR`` or environment
+variable of the same name. If the config file cannot be found, the libraries are
+heuristically searched by names and paths in ``TAICHI_C_API_INSTALL_DIR``.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module provides the following imported targets, if found:
+
+``taichi::runtime``
+  The Taichi Runtime library.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``taichi_FOUND``
+  True if a Taichi installation is found.
+``taichi_VERSION``
+  Version of installed Taichi. Components might have lower version numbers.
+``taichi_INCLUDE_DIRS``
+  Paths to Include directories needed to use Taichi.
+``taichi_LIBRARIES``
+  Paths to Taichi linking libraries (``.libs``).
+``taichi_REDIST_LIBRARIES``
+  Paths to Taichi redistributed runtime libraries (``.so`` and ``.dll``). You
+  might want to copy them next to your executables.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``taichi_runtime_VERSION``
+  Taichi runtime library version.
+``taichi_runtime_INCLUDE_DIR``
+  The directory containing ``taichi/taichi.h``.
+``taichi_runtime_LIBRARY``
+  Path to linking library of ``taichi_runtime``.
+``taichi_runtime_REDIST_LIBRARY``
+  Path to redistributed runtime library of ``taichi_runtime``.
+
+#]=======================================================================]
+
+cmake_policy(PUSH)
+
+# Support `IN_LIST` in CMake `if` command.
+if(POLICY CMP0057)
+    cmake_policy(SET CMP0057 NEW)
+endif()
+
+# Find TiRT in the installation directory. The installation directory is
+# traditionally specified by an environment variable
+# `TAICHI_C_API_INSTALL_DIR`.
+if(NOT EXISTS ${TAICHI_C_API_INSTALL_DIR})
```

Comment:
Will `FindTaichi.cmake` get shipped along with C-API? If so,  ${TAICHI_C_API_INSTALL_DIR} should be a relative path to `FindTaichi.cmake` (like `./..`?)

Reply:
This one should be set up by the user. This FindTaichi.cmake is expected to be copied by the user to their repositories, in a folder called `cmake`. Then, when the user builds their project, CMake can find that package using this script following the rules in `https://cmake.org/cmake/help/v3.0/command/find_package.html#command:find_package`.